### ISSUE TITLE
using GAP does not see packages in ~/.gap/pkg

### ISSUE THREAD

**Issue by mohamed-barakat** [oscar-system/GAP.jl#340]
I have the package `ZariskiFrames` and all its dependencies under `~/.gap/pkg`. Indeed
```
$ ~/.julia/packages/GAP/ABCDE/gap.sh
gap> LoadPackage( "ZariskiFrames" );
true
```
However `using GAP` does not see the packages
```
julia> using GAP
Adding path ~/.julia/packages/GAP/ABCDE/gap/.libs to DL_LOAD_PATH
 ┌───────┐   GAP 4.dev of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin19.3.0-julia64-kv7
 Configuration:  gmp 6.1.2, Julia GC, Julia 1.2.0, readline
 Loading the library and packages ...
 Packages:   GAPDoc 1.6.3, PrimGrp 3.4.0, SmallGrp 1.4.1, TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'

julia> LoadPackageAndExposeGlobals( "ZariskiFrames", Main, all_globals = true )
ERROR: cannot load package ZariskiFrames
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] #LoadPackageAndExposeGlobals#9(::Bool, ::Bool, ::typeof(LoadPackageAndExposeGlobals), ::String, ::Module) at ~/.julia/packages/GAP/C70xu/src/packages.jl:47
 [3] (::getfield(GAP, Symbol("#kw##LoadPackageAndExposeGlobals")))(::NamedTuple{(:all_globals,),Tuple{Bool}}, ::typeof(LoadPackageAndExposeGlobals), ::String, ::Module) at ./none:0
 [4] top-level scope at REPL[2]:1
```

**Comment by ThomasBreuer** [comment]
@mohamed-barakat Yes, currently Julia starts its GAP with the `-r` option, see #284 .
This means that the contents of the user's `~/.gap` directory is ignored.

For the moment, you can call `GAP.Globals.ExtendRootDirectories( GAP.julia_to_gap( [ GAP.Globals.UserHomeExpand( g"~/.gap" ) ] ) )`,
afterwards `LoadPackageAndExposeGlobals` should work.

I will create a pull request, in order to omit the `-r` option when starting GAP.

**Comment by ThomasBreuer** [comment]
fixed by #341

**Comment by mohamed-barakat** [comment]
@ThomasBreuer: Ah, I was unaware of the GAP command `ExtendRootDirectories`. Excellent, thank you very much.

**Comment by mohamed-barakat** [comment]
@fingolfin: Thank you very much for the fix by #341

**Comment by mohamed-barakat** [comment]
> For the moment, you can call `GAP.Globals.ExtendRootDirectories( GAP.julia_to_gap( [ GAP.Globals.UserHomeExpand( g"~/.gap" ) ] ) )`,
> afterwards `LoadPackageAndExposeGlobals` should work.
> 
> I will create a pull request, in order to omit the `-r` option when starting GAP.

For the record: I tested it with another directory and it did not. What is missing is a trailing slash. So this works
```
GAP.Globals.ExtendRootDirectories( GAP.julia_to_gap( [ GAP.Globals.UserHomeExpand( g"~/.gap/" ) ] ) )
```
